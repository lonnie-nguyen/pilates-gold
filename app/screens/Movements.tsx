import {View, Text, TouchableOpacity, StyleSheet} from 'react-native';
import React from 'react';
import Menu from "../Components/Menu";

//This is Movement page(Home Page)
const MovementScreen = () => {
    return (
        <View style={styles.container}>
            <Text style={styles.screenTitle}>Movements</Text>
            <Menu />
        </View>
    );
};

const styles = StyleSheet.create({
    container: {
        padding: 16,
        marginTop: 24,
    },
    screenTitle: {
        fontSize: 24,
        marginTop: 8,
        fontWeight: 'bold',
    },
});

export default MovementScreen;