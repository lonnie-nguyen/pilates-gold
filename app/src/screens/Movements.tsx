import {View, Text, StyleSheet, FlatList} from 'react-native';
import React, {useEffect, useState} from 'react';
import {collection, getDocs, query} from "firebase/firestore";
import {db} from "../../firebaseConfig";
import { getStorage, ref, getDownloadURL } from "firebase/storage";
import ExerciseCard from "../Components/ExerciseCardComp";

export default class MovementScreen extends React.Component<any, any> {
    constructor(props: { title: string}) {
        super(props);
        this.state = {
            data: {},
            imageURLs: {},
        };
    }

    componentDidMount() {
        const storage = getStorage()
        // const q = query(collection(db, "movements"))
        const q = query(collection(db, "weekly-moves"))
        const setDatabase = async () => {
            const querySnapshot = await getDocs(q);
            querySnapshot.forEach((doc) => {
                this.setState((prevState: { data: any; }) => ({
                    data: {...prevState.data, [doc.id]: doc.data()},
                }))
                const value = doc.data()
                if (value.imageURL) {
                    const picRef = ref(storage, value.imageURL)
                    getDownloadURL(picRef)
                        .then((url) => {
                            this.setState((prevState: { imageURLs: any; }) => ({
                                imageURLs: {...prevState.imageURLs, [doc.id]: url},
                            }))
                        })
                        .catch((error) => {
                            switch (error.code) {
                                case 'storage/object-not-found':
                                    break;
                                case 'storage/unauthorized':
                                    break;
                                case 'storage/canceled':
                                    break;
                                case 'storage/unknown':
                                    break;
                            }
                        });
                }
            });

        }
        setDatabase();

    }

    ListCard = () => {
        const items = [];
        for (const [key, value] of Object.entries(this.state.data)) {
            items.push(
                {title : this.state.data[key].title, url : this.state.imageURLs[key], screenName : "MovementInfo"}
            );
        }
        return items;
    }

    renderCard = ({item}: {item: any}) => {
        return (
          <ExerciseCard screenName={item.screenName} imageAddr={item.url} name={item.title} />
        )
    }

    render () {
        const listData = this.ListCard();
        return (
            <View style={styles.movementsView}>
                <FlatList showsVerticalScrollIndicator={false} 
                          columnWrapperStyle={styles.row} 
                          numColumns={2}  
                          data={listData} 
                          renderItem={this.renderCard}/>
            </View>
        );
    }
}

const styles = StyleSheet.create({
    movementsView: {
      position: "relative",
      backgroundColor: "#f9eebf",
      flex: 1,
      paddingLeft: 10,
      paddingRight:10,
      flexDirection: "column",
      alginItems: "center",
      justifyContent: "flex-start",
    },
    row: {
      flex: 1,
      marginTop: 10,
      alignItems: "flex-start",
      justifyContent: "space-between",
    }
});
