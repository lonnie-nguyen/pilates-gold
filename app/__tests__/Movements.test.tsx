import React from 'react'
import { it, expect } from '@jest/globals'
import { render } from '@testing-library/react-native'
import Movements from '../src/screens/Movements'

it('movements page snapshot', () => {
  const { toJSON } = render(<Movements />)
  expect(toJSON()).toMatchSnapshot()
})
