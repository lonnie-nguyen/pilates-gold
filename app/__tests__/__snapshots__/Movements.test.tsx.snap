// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`homwpage snapshot 1`] = `
<View
  style={
    {
      "alginItems": "center",
      "backgroundColor": "#D9EBF2",
      "flex": 1,
      "flexDirection": "column",
      "justifyContent": "flex-start",
      "paddingLeft": 10,
      "paddingRight": 10,
      "position": "relative",
    }
  }
>
  <RCTScrollView
    data={[]}
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    showsVerticalScrollIndicator={false}
    stickyHeaderIndices={[]}
    viewabilityConfigCallbackPairs={[]}
  >
    <View />
  </RCTScrollView>
</View>
`;

exports[`movements page snapshot 1`] = `
<View
  style={
    {
      "alginItems": "center",
      "backgroundColor": "#D9EBF2",
      "flex": 1,
      "flexDirection": "column",
      "justifyContent": "flex-start",
      "paddingLeft": 10,
      "paddingRight": 10,
      "position": "relative",
    }
  }
>
  <RCTScrollView
    data={[]}
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    showsVerticalScrollIndicator={false}
    stickyHeaderIndices={[]}
    viewabilityConfigCallbackPairs={[]}
  >
    <View />
  </RCTScrollView>
</View>
`;
